plugin-core/src/main/java/appland/installGuide/analyzer/languages/JavaScriptLanguageAnalyzer.java

```diff
@@ -161,9 +164,40 @@ private void loadPackageLock(@NotNull VirtualFile directory, List<Pair<String, S
     }
 
     /**
-     * Reads YAML file "yarn.lock".
+     * Reads file "yarn.lock" of Yarn v1, which isn't valid YAML but a custom format.
+     * We're implementing a very simple lookup to avoid writing a parser.
      */
-    private void loadYarnLock(@NotNull VirtualFile directory, List<Pair<String, String>> target) {
+    private void loadYarn1Lock(@NotNull VirtualFile directory, List<Pair<String, String>> target) {
+        var lockFile = directory.findChild("yarn.lock");
+        if (lockFile == null) {
+            return;
+        }
+
+        var document = FileDocumentManager.getInstance().getDocument(lockFile);
+        if (document == null) {
+            return;
+        }
+
+        // Dependencies are
+        //    "mocha@~10.2.0":
+        // or
+        //    mocha@~10.2.0:
+
+        var dependencyPattern = Pattern.compile("^\"?(.+?)@(.+?)\"?:$");
```

Comment:
That's a good idea :) It does detect the requested instead of the resolved version, but this shouldn't matter all that much.